[TST] Upcoming dependency test failures

The weekly build with nightly wheels from numpy and pandas
has failed. Check the logs for any updates that need to be
made in matplotlib.
https://github.com/matplotlib/matplotlib/actions/runs/4837230527

Likely related to numpy/numpy#23399

```
 FAILED lib/matplotlib/tests/test_axes.py::test_mollweide_forward_inverse_closure - AssertionError: 
Arrays are not almost equal to 3 decimals

Mismatched elements: 720 / 129600 (0.556%)
Max absolute difference: 3.14159264
Max relative difference: 2.4335443e+08
 x: array([[-3.142, -1.571],
       [-3.124, -1.571],
       [-3.107, -1.571],...
 y: array([[-1.291e-08, -1.571e+00],
       [-1.284e-08, -1.571e+00],
       [-1.277e-08, -1.571e+00],...
FAILED lib/matplotlib/tests/test_polar.py::test_cursor_precision - AssertionError: assert 'θ=0.0π (0°), r=0.000' == 'θ=0π (0°), r=0.000'
  - θ=0π (0°), r=0.000
  + θ=0.0π (0°), r=0.000
  ?    ++
FAILED lib/mpl_toolkits/mplot3d/tests/test_axes3d.py::test_bar3d_lightsource - AssertionError: 
Arrays are not equal

Mismatched elements: 36 / 48 (75%)
Max absolute difference: 2.22044605e-16
Max relative difference: 2.94612669e-16
 x: array([[0.229806, 0.298718, 0.753683, 1.      ],
       [0.328604, 0.439712, 0.869587, 1.      ],
       [0.435815, 0.570707, 0.951717, 1.      ],...
 y: array([[0.229806, 0.298718, 0.753683, 1.      ],
       [0.328604, 0.439712, 0.869587, 1.      ],
       [0.435815, 0.570707, 0.951717, 1.      ],...
```

First test is puzzling, it is intended to be a round drip of a transform and its inverse, some values seem to be off by precisely pi

Second test is a string comparison where the value for theta. My guess (based on following assertions/comments from that test) is that one of the input 0's is passed through a sin/cos and gets a small non-zero value as a result, which triggers an increase in the precision of theta.

Last one is a very small deviation (looks to be epsilon, i.e. `np.spacing(1)`) can maybe be compensated for with a tight but finite precision instead of strict equality. Seemingly due to `sin(90deg)` (the lightsource angle in the test) not being _precisely_ 1. (note on current released numpy, sin(np.pi) is not actually precisely 0)

All three tests do deal with angles/computations stemming from sin/cos so probably all the same root cause.

--- a/lib/matplotlib/tests/test_axes.py
+++ b/lib/matplotlib/tests/test_axes.py
@@ -4415,7 +4415,8 @@ def test_mollweide_forward_inverse_closure():
 
     # set up 1-degree grid in longitude, latitude
     lon = np.linspace(-np.pi, np.pi, 360)
-    lat = np.linspace(-np.pi / 2.0, np.pi / 2.0, 180)
+    # The poles are degenerate and thus sensitive to floating point precision errors
+    lat = np.linspace(-np.pi / 2.0, np.pi / 2.0, 180)[1:-1]
     lon, lat = np.meshgrid(lon, lat)
     ll = np.vstack((lon.flatten(), lat.flatten())).T
 

--- a/lib/matplotlib/tests/test_polar.py
+++ b/lib/matplotlib/tests/test_polar.py
@@ -425,13 +425,13 @@ def test_axvline_axvspan_do_not_modify_rlims():
 def test_cursor_precision():
     ax = plt.subplot(projection="polar")
     # Higher radii correspond to higher theta-precisions.
-    assert ax.format_coord(0, 0) == "θ=0π (0°), r=0.000"
+    assert ax.format_coord(0, 0.005) == "θ=0.0π (0°), r=0.005"
     assert ax.format_coord(0, .1) == "θ=0.00π (0°), r=0.100"
     assert ax.format_coord(0, 1) == "θ=0.000π (0.0°), r=1.000"
-    assert ax.format_coord(1, 0) == "θ=0.3π (57°), r=0.000"
+    assert ax.format_coord(1, 0.005) == "θ=0.3π (57°), r=0.005"
     assert ax.format_coord(1, .1) == "θ=0.32π (57°), r=0.100"
     assert ax.format_coord(1, 1) == "θ=0.318π (57.3°), r=1.000"
-    assert ax.format_coord(2, 0) == "θ=0.6π (115°), r=0.000"
+    assert ax.format_coord(2, 0.005) == "θ=0.6π (115°), r=0.005"
     assert ax.format_coord(2, .1) == "θ=0.64π (115°), r=0.100"
     assert ax.format_coord(2, 1) == "θ=0.637π (114.6°), r=1.000"
 

--- a/lib/mpl_toolkits/mplot3d/tests/test_axes3d.py
+++ b/lib/mpl_toolkits/mplot3d/tests/test_axes3d.py
@@ -200,9 +200,9 @@ def test_bar3d_lightsource():
 
     # Testing that the custom 90° lightsource produces different shading on
     # the top facecolors compared to the default, and that those colors are
-    # precisely the colors from the colormap, due to the illumination parallel
-    # to the z-axis.
-    np.testing.assert_array_equal(color, collection._facecolor3d[1::6])
+    # precisely (within floating point rounding errors of 4 ULP) the colors
+    # from the colormap, due to the illumination parallel to the z-axis.
+    np.testing.assert_array_max_ulp(color, collection._facecolor3d[1::6], 4)
 
 
 @mpl3d_image_comparison(['contour3d.png'], style='mpl20')

